lib/thread.c

```diff
@@ -44,13 +44,37 @@ DEFINE_MTYPE_STATIC(LIB, THREAD_STATS,  "Thread stats")
 /* Relative time, since startup */
 static struct hash *cpu_record = NULL;
 
-static unsigned long
+unsigned long
 timeval_elapsed (struct timeval a, struct timeval b)
 {
   return (((a.tv_sec - b.tv_sec) * TIMER_SECOND_MICRO)
 	  + (a.tv_usec - b.tv_usec));
 }
 
+void
+timeval_copy (struct timeval *a, struct timeval b)
+{
+  if (!a)
+    return;
+  a->tv_sec = b.tv_sec;
+  a->tv_usec = b.tv_usec;
+}
+
+
+struct timeval
+timeval_add_msec (struct timeval a, time_t msec)
```

Comment:
This is essentially a reimplementation of timeval_adjust, which was removed per #104. Worth splitting out into it's own PR.